tests/Tests/400__ods_dashboard/400__ods_dashboard.robot

```diff
@@ -155,7 +155,88 @@ Verify "Notebook Images Are Building" Is Not Shown When No Images Are Building
     Wait Until All Builds Are Complete  namespace=redhat-ods-applications
     RHODS Notification Drawer Should Not Contain  message=Notebooks images are building
 
+Verify Favorite Resource Cards
+    [Tags]    ODS-389
+    Click Link    Resources
+    Sleep    5s
+   ${list_of_tile} =    Get List Of Ids Of Tiles
+   Verify Star Icons Are Clickable    ${list_of_tile}
+
+    FOR  ${index}  IN RANGE    27    48
+         Click Element    //*[@id="${list_of_tile}[${index}]"]/div[1]/span
+         ${clicked} =     Get Element Attribute    //*[@id="${list_of_tile}[${index}]"]/div[1]/span    class
+         Should Be Equal    ${clicked}    odh-dashboard__favorite m-is-favorite
+    END
+
+    ${list_of_new_tile} =    Get List Of Ids Of Tiles
+
+    Lists Should Be Equal    ${list_of_tile}    ${27}    ${48}
+    ...                      ${list_of_new_tile}    ${0}    ${21}
+
+    Click Button    //*[@id="list-view"]
+    Sleep    5s
+
+
+    ${list_of_new_tile} =    Get List Of Atrributes    //div[@class="odh-list-item__doc-title"]    id
+
+    Lists Should Be Equal    ${list_of_new_tile}    ${0}    ${42}
+    ...                      ${list_of_tile}    ${27}    ${48}   ${2}
+
+    Click Button    //*[@id="card-view"]
+    Sleep    2s
+    FOR  ${index}  IN RANGE    27    48
+         Log To Console    ${index}    ${list_of_tile}[${index}]
+         Click Element    //*[@id="${list_of_tile}[${index}]"]/div[1]/span
+         ${clicked} =     Get Element Attribute    //*[@id="${list_of_tile}[${index}]"]/div[1]/span    class
+    END
+
+
 *** Keywords ***
+Verify Star Icons Are Clickable
+    [Documentation]    Verifies that star icons in the resources page are clickable
+    [Arguments]    ${list_of_ids}
+    FOR    ${element}    IN    @{list_of_ids}
+         ${not_clicked} =    Get Element Attribute    //*[@id="${element}"]/div[1]/span    class
+         Should Be Equal    ${not_clicked}    odh-dashboard__favorite
+         Click Element    //*[@id="${element}"]/div[1]/span
```

Comment:
this could become a kw like "Set Item As Favorite"

Reply:
Done